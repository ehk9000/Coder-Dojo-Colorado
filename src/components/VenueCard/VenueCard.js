import React from 'react';

function VenueCard({ venue }) {
  const { 
    id, name, notes, addresses 
  } = venue;
  return (
    <article className="venue-card" key={id}>
      <h3>
        {name}
      </h3>
      <p>{notes}</p>
    </article> 
  );
}

export default VenueCard;
