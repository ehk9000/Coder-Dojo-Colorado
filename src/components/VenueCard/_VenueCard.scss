.venue-card {
  background-color: #fff;
  padding: 50px;
  margin: 10px;
  display: flex;

  .dojo-logo {
    width: 40px;
    padding: 0;
  }

  h3 {
    color: #298244;
    font-size: 1.4rem;
    font-weight: 500;
  }

  p {
    color: #999999;
  }

  .info-container {
    margin: 0 20px;
  }
}