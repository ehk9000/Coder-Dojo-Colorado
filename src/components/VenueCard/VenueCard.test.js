import React from 'react';
import { shallow } from 'enzyme';
import VenueCard from './VenueCard';

describe('VenueCard', () => {
  let wrapper;
  const mockVenue = {
    id: 1,
    name: 'Venue One',
    notes: 'Here be Dragons',
    addresses: [
      {
        city: 'Denver',
        street1: '567 Main St.',
        street2: 'Apt 678',
        zip: '80128',
        state: 'CO',
        id: 4
      }
    ]
  };

  beforeEach(() => {
    wrapper = shallow(<VenueCard venue={mockVenue} key={mockVenue.id} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
