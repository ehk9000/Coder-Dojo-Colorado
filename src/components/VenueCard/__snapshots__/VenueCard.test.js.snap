// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueCard should match snapshot 1`] = `
<Link
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/venues/1/events"
>
  <article
    className="venue-card"
    key="1"
  >
    <ForwardRef(SvgYinYangLogo)
      className="dojo-logo"
    />
    <div
      className="info-container"
    >
      <h3>
        Venue One
      </h3>
      <p
        key="4"
      >
        567 Main St., Apt 678 Denver, CO 80128
      </p>
    </div>
  </article>
</Link>
`;
